numpy/_core/tests/test_array_coercion.py

```diff
@@ -636,7 +636,7 @@ def __len__(self):
                 obj[0][0] = 2  # replace with a different list.
                 raise ValueError("not actually a sequence!")
 
-            def __getitem__(self):
+            def __getitem__(self):  # noqa: PLE0302
```

Comment:
The `__len__` will be called first, so

```suggestion
            def __getitem__(self, _, /):
```

should also work